earth2studio/models/px/aifs.py

```diff
@@ -0,0 +1,857 @@
+
+        # Get cos, sin of latitude and longitude
+        # (cos_latitude, sin_latitude, cos_longitude, sin_longitude)
+        cos_latitude = torch.cos(torch.deg2rad(self.latitudes))
+        sin_latitude = torch.sin(torch.deg2rad(self.latitudes))
+        cos_longitude = torch.cos(torch.deg2rad(self.longitudes))
+        sin_longitude = torch.sin(torch.deg2rad(self.longitudes))
+        cos_latitude = torch.cat([cos_latitude, cos_latitude], dim=1)
+        cos_longitude = torch.cat([cos_longitude, cos_longitude], dim=1)
+        sin_latitude = torch.cat([sin_latitude, sin_latitude], dim=1)
+        sin_longitude = torch.cat([sin_longitude, sin_longitude], dim=1)
+
+        # Get cos, sin of Julian day
+        cos_julian_day_0, sin_julian_day_0 = self.get_cos_sin_julian_day(
+            coords["time"][0] - np.timedelta64(6, "h"), self.longitudes
+        )
+        cos_julian_day_1, sin_julian_day_1 = self.get_cos_sin_julian_day(
+            coords["time"][0], self.longitudes
+        )
+        cos_julian_day = torch.cat([cos_julian_day_0, cos_julian_day_1], dim=1)
+        sin_julian_day = torch.cat([sin_julian_day_0, sin_julian_day_1], dim=1)
+
+        # Get cos, sin local time
+        cos_local_time_0, sin_local_time_0 = self.get_cos_sin_local_time(
+            coords["time"][0] - np.timedelta64(6, "h"), self.longitudes
+        )
+        cos_local_time_1, sin_local_time_1 = self.get_cos_sin_local_time(
+            coords["time"][0], self.longitudes
+        )
+        cos_local_time = torch.cat([cos_local_time_0, cos_local_time_1], dim=1)
+        sin_local_time = torch.cat([sin_local_time_0, sin_local_time_1], dim=1)
+
+        # Get cosine zenith angle
+        # Add insolation / cosine zenith angle
+        cos_zenith_angle_0 = self.get_cosine_zenith_fields(
+            coords["time"][0] - np.timedelta64(6, "h"), self.latitudes, self.longitudes
+        )
+        cos_zenith_angle_1 = self.get_cosine_zenith_fields(
+            coords["time"][0], self.latitudes, self.longitudes
+        )
+        cos_zenith_angle = torch.cat([cos_zenith_angle_0, cos_zenith_angle_1], dim=1)
+
+        # Combine inputs
+        x = torch.cat(
+            [
+                x[:, :, :, :90],
+                cos_latitude.repeat(shape[0] * shape[1], 1, 1, 1),
+                cos_longitude.repeat(shape[0] * shape[1], 1, 1, 1),
+                sin_latitude.repeat(shape[0] * shape[1], 1, 1, 1),
+                sin_longitude.repeat(shape[0] * shape[1], 1, 1, 1),
+                cos_julian_day.repeat(shape[0] * shape[1], 1, 1, 1),
+                cos_local_time.repeat(shape[0] * shape[1], 1, 1, 1),
+                sin_julian_day.repeat(shape[0] * shape[1], 1, 1, 1),
+                sin_local_time.repeat(shape[0] * shape[1], 1, 1, 1),
+                cos_zenith_angle.repeat(shape[0] * shape[1], 1, 1, 1),
+                x[:, :, :, 90:],
+            ],
+            dim=3,
+        )
+        # ], dim=3).to(torch.bfloat16)
+
+        return x
```

Comment:
x here is  of shape `x.reshape([shape[0] * shape[1], shape[2], shape[3] + n, -1])`  does it make sense to transform it back into the ([shape[0], shape[1], shape[2], shape[3], shape[4], shape[5]])` at the end of this?